lnwire/extra_bytes.go

```diff
@@ -60,6 +60,17 @@ func (e *ExtraOpaqueData) Decode(r io.Reader) error {
 		*e = make([]byte, 0)
 	}
 
+	tlvStream, err := tlv.NewStream()
```

Comment:
I've left this in the commit since this is what helped expose the bug. 

We need this as part of the graph SQL work and i've confirmed that a fully synced LND node today has valid TLV extra blobs only. 

But let me know if ya'll would prefer this not to be in the PR

Reply:
I see how this is required for graph stuff. But is `ExtraOpaqueData` the correct place for this check? Because it looks like we are using that struct in many many different messages (channel negotiation, dynamic commitments, custom channel HTLCs, onion errors, just to name a few). Are we sure that **all** those messages need to have valid data?
Perhaps we can. But to make sure we don't introduce a showstopper perhaps we should move this check out of `ExtraOpaqueData` and into the messages we really need to be correct for the graph stuff?
Or we can turn it into a method `EnsureTLV` that is then called from the different places?
I know that's a bit more code duplication and involves more changes.
But like this I think it's a bit too risky.

Reply:
ok yeah - good idea! will update :+1: